### PR TITLE
[Task][75153] Create UI Actor screen & trailer screen

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 75153: Create UI Actor screen & trailer screen](https://edu-redmine.sun-asterisk.vn/issues/75153)
## WHAT
- Add logic remove movie in favorite screen , 
- Add action tapped cell to detail movie
- Create ui Actor screen
- Create ui trailer screen
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: MovieHub/MovieHub/Scenes/Actor/ActorInfo/ActorInfoCell.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+final class ActorInfoCell: UITableViewCell, NibReusable {
+
+    @IBOutlet private weak var containerView: UIView!
+    @IBOutlet private weak var backdropImage: UIImageView!
+    @IBOutlet private weak var actorImage: UIImageView!
+    @IBOutlet private weak var actorName: UILabel!
+    @IBOutlet private weak var actorBio: UILabel!
+    @IBOutlet private weak var actorKnownAs: UILabel!
+    @IBOutlet private weak var actorDate: UILabel!
+    @IBOutlet private weak var actorPlace: UILabel!
```
Comment (ghost) [inline MovieHub/MovieHub/Scenes/Actor/ActorInfo/ActorInfoCell.swift]: thêm label đằng sau nữa e

File: MovieHub/MovieHub/Scenes/Actor/ActorMovie/ActorMovieCell.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    @IBOutlet private weak var titleLabel: UILabel!
+    @IBOutlet private weak var collectionView: UICollectionView!
+    @IBOutlet private weak var pageView: UIPageControl!
+    
+    var movies = [Movie]() {
+        didSet {
+            collectionView.reloadData()
+        }
+    }
```
Comment (ghost) [inline MovieHub/MovieHub/Scenes/Actor/ActorMovie/ActorMovieCell.swift]: dùng trong viewcontroller thì private lại e

File: MovieHub/MovieHub/Scenes/Actor/ActorViewController.swift [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import Reusable
+
+final class ActorViewController: UIViewController, NibReusable {
+
+    @IBOutlet private weak var tableView: UITableView!
+    
+    private var movieRepository: MovieRepositoryType = MovieRepository()
+    var idActor: Int?
+    private var actor = Actor()
+    private var heightTableViewConstraint = 500.0
```
Comment (ghost) [inline MovieHub/MovieHub/Scenes/Actor/ActorViewController.swift]: nên tạo 1 enum constant của từng viewcontroller để chứa những cái như này

File: MovieHub/MovieHub/Scenes/Actor/ActorViewController.swift [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let section = ActorSectionType(rawValue: indexPath.section) else {
+            return UITableViewCell()
+        }
+        switch section {
+
```
Comment (ghost) [inline MovieHub/MovieHub/Scenes/Actor/ActorViewController.swift]: remove blank

File: MovieHub/MovieHub/Scenes/Actor/ActorViewController.swift [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            let cell = tableView.dequeueReusableCell(for: indexPath, cellType: ActorInfoCell.self)
+            cell.configCell(actor: actor)
+            return cell
+            
+        case .movies:
+            let cell = tableView.dequeueReusableCell(for: indexPath, cellType: ActorMovieCell.self)
+            cell.configCell(movies: actor.movieCredits?.movies ?? [])
+            cell.tappedMovie = { [weak self] movie in
+                guard let self else { return }
+                guard let movieID = movie.id else { return }
```
Comment (ghost) [inline MovieHub/MovieHub/Scenes/Actor/ActorViewController.swift]: gộp lại e

File: MovieHub/MovieHub/Scenes/Favorite/FavoriteViewController.swift [hunk 6]
```diff
@@ -67,6 +67,27 @@ extension FavoriteViewController: UITableViewDelegate, UITableViewDataSource {
...
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let movieID = dataSource[indexPath.row].id
+        toMovieDetailScreen(movieID: Int(movieID))
+    }
+    
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+          if editingStyle == .delete {
+              print("---------")
+          }
+      }
```
Comment (ghost) [inline MovieHub/MovieHub/Scenes/Favorite/FavoriteViewController.swift]: k dùng thì xoá e

File: MovieHub/MovieHub/Scenes/Home/HomeViewController.swift [hunk 7]
```diff
@@ -176,7 +176,8 @@ extension HomeViewController: UITableViewDelegate, UITableViewDataSource {
             cell.configCell(movies: movies)
             cell.tappedMovie = { [weak self] movie in
                 guard let self else { return }
-                self.toMovieDetailScreen(movie: movie)
+                guard let movieID = movie.id else { return}
```
Comment (ghost) [inline MovieHub/MovieHub/Scenes/Home/HomeViewController.swift]: gộp lại unwrap chung với line 178

File: MovieHub/MovieHub/Models/Actor.swift [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+struct Actor: Codable {
+    var id: Int?
+    var name: String?
+    var birthday: String?
+    var place: String?
+    var profilePath: String?
+    var biography: String?
+    var knownFor: String?
+    var movieCredits : MovieCredits?
+    enum CodingKeys: String, CodingKey {
```
Comment (dongtg-1485) [inline MovieHub/MovieHub/Models/Actor.swift:20]: break line

File: MovieHub/MovieHub/Models/Actor.swift [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        case birthday
+        case place = "place_of_birth"
+        case profilePath = "profile_path"
+        case biography
+        case knownFor = "known_for_department"
+        case movieCredits = "movie_credits"
+    }
+}
+
+struct MovieCredits: Codable {
```
Comment (dongtg-1485) [inline MovieHub/MovieHub/Models/Actor.swift]: tách file

File: MovieHub/MovieHub/Scenes/Actor/ActorInfo/ActorInfoCell.swift [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    private func configView() {
+        actorBioLabel.numberOfLines = 4
+        actorImageView.layer.cornerRadius = 12
+        actorImageView.layer.cornerCurve = .continuous
+        actorImageView.layer.borderColor = UIColor.white.cgColor
+        actorImageView.layer.borderWidth = 3
+        backdropImageView.addGradientOverlay()
+    }
+    
+    func configCell(actor: Actor) {
```
Comment (dongtg-1485) [inline MovieHub/MovieHub/Scenes/Actor/ActorInfo/ActorInfoCell.swift]:
```suggestion
    func setContentForCell(actor: Actor) {
```

File: MovieHub/MovieHub/Scenes/Favorite/FavoriteViewController.swift [hunk 11]
```diff
@@ -67,6 +64,36 @@ extension FavoriteViewController: UITableViewDelegate, UITableViewDataSource {
...
         return cell
     }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let movieID = dataSource[indexPath.row].id
+        toMovieDetailScreen(movieID: Int(movieID))
+    }
+    
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
```
Comment (dongtg-1485) [inline MovieHub/MovieHub/Scenes/Favorite/FavoriteViewController.swift]: dùng switch case cho enum thay vì if else nhé

File: MovieHub/MovieHub/Utils/Enum.swift [hunk 12]
```diff
@@ -51,3 +51,9 @@ enum MovieSectionType: Int {
     case similar
     case total
 }
+
+enum ActorSectionType: Int {
```
Comment (dongtg-1485) [inline MovieHub/MovieHub/Utils/Enum.swift]: tách hết file ra nhé

**Review by ghost — APPROVED** [review]

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**